docs/config/java.md

```diff
@@ -1,17 +1,65 @@
 ﻿[← Back Home](../README.md)
 
+# Configuring Java Services
+
+Spring Boot uses two main YAML configuration files:
+
+* `bootstrap.yml`
+    * Loaded before application.yml during the startup process.
+    * Used for early-stage configuration that is required before the main application context is loaded.
+* `application.yml`
+    * Loaded after bootstrap.yml, within the main application context.
+
+Some properties may want to be put in environment-specific configuration files with the naming convention `bootstrap-ENV.yml` and `application-ENV.yml`. For example, there may be `bootstrap-dev.yml` and `bootstrap-prod.yml`. At deployment-time, you can specify an environment variable `SPRING_PROFILES_ACTIVE` to indicate whether to load `dev` or `prod` on top of the default configurations.
+
+When deploying Spring Boot applications in Docker/Kubernetes, overriding configuration using environment variables is a common practice. Environment variables override JSON/YML specified in application config files. Here's some guidance on how to represent the hierarchy JSON/YML structure in an environment variable:
+
+* Replace dots (.) and dashes (-) with underscores (_).
+* Convert all characters to uppercase.
+* Reference array indexes using _INDEX_ notation (zero-based).
+* Remove any special characters (like hyphens) but keep underscores.
+* Maintain nesting structure by replacing separators (.) with _.
+* For camelCase keys, convert them to snake_case before applying rules.
+* Do not add extra underscores at the beginning or end.
+* Do not enclose values in quotes unless required.
+* For boolean values (true/false), use standard conversion (TRUE/FALSE if required).
+* Use SPRING_APPLICATION_JSON if passing deeply nested structures in a single variable.
+
+Examples of environment variable naming conventions:
+
+| YAML / JSON Key                                                        | Converted Environment Variable                                      |
+|------------------------------------------------------------------------|---------------------------------------------------------------------|
+| server.port                                                            | SERVER_PORT                                                         |
+| spring.datasource.url                                                  | SPRING_DATASOURCE_URL                                               |
+| spring.cloud.azure.appconfiguration.stores[0].selects[0].label-filter	 | SPRING_CLOUD_AZURE_APPCONFIGURATION_STORES_0_SELECTS_0_LABEL_FILTER |
+| my.custom.settings[2].api-key                                          | MY_CUSTOM_SETTINGS_2_API_KEY                                        |
+| config.services[1].endpoints.internal-url                              | CONFIG_SERVICES_1_ENDPOINTS_INTERNAL_URL                            |
```

Comment:
According to the "official" rules, these should be `LABELFILTER`, `APIKEY` and `INTERNALURL`, though as I'm sure you've found, it seems to work either way.